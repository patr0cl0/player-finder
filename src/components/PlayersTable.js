import React from 'react';
import PropTypes from 'prop-types';
import {
  Table,
  TableHead,
  TableCell,
  TableRow,
  TableBody,
} from '@material-ui/core';

const PlayersTable = ({ data }) => (
  <Table>
    <TableHead>
      <TableRow>
        <TableCell>Player</TableCell>
        <TableCell>Position</TableCell>
        <TableCell>Age</TableCell>
      </TableRow>
    </TableHead>
    <TableBody>
      {data.map(player => (
        <TableRow key={player.name + player.position} className="PlayersTable--row">
          <TableCell>{player.name}</TableCell>
          <TableCell>{player.position}</TableCell>
          <TableCell>{player.age}</TableCell>
        </TableRow>
      ))}
    </TableBody>
  </Table>
);

PlayersTable.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    position: PropTypes.string,
    age: PropTypes.number,
  })).isRequired,
};


export default PlayersTable;
