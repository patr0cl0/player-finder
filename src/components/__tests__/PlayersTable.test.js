import React from 'react';
import { shallow } from 'enzyme';
import PlayersTable from '../PlayersTable';

const playersTableMockProps = {
  data: [{
    contractUntil: '2022-06-30',
    dateOfBirth: '1993-05-13',
    jerseyNumber: 9,
    name: 'Romelu Lukaku',
    nationality: 'Belgium',
    position: 'Centre-Forward',
  }, {
    contractUntil: '2019-06-30',
    dateOfBirth: '1990-11-07',
    jerseyNumber: 1,
    name: 'David de Gea',
    nationality: 'Spain',
    position: 'Keeper',
  }],
};

describe('PlayersTable tests', () => {
  it('should match the snapshot', () => {
    const component = shallow(<PlayersTable {...playersTableMockProps} />);

    expect(component).toMatchSnapshot();
  });

  it('should render two rows', () => {
    const component = shallow(<PlayersTable {...playersTableMockProps} />);

    const rows = component.find('.PlayersTable--row');

    expect(rows.length).toBe(2);
  });
});
