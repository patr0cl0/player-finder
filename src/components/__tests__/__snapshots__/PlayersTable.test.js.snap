// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersTable tests should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayersTable
    data={
      Array [
        Object {
          "contractUntil": "2022-06-30",
          "dateOfBirth": "1993-05-13",
          "jerseyNumber": 9,
          "name": "Romelu Lukaku",
          "nationality": "Belgium",
          "position": "Centre-Forward",
        },
        Object {
          "contractUntil": "2019-06-30",
          "dateOfBirth": "1990-11-07",
          "jerseyNumber": 1,
          "name": "David de Gea",
          "nationality": "Spain",
          "position": "Keeper",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(TableHead)>
          <WithStyles(TableRow)>
            <WithStyles(TableCell)>
              Player
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Position
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Age
            </WithStyles(TableCell)>
          </WithStyles(TableRow)>
        </WithStyles(TableHead)>,
        <WithStyles(TableBody)>
          <WithStyles(TableRow)
            className="PlayersTable--row"
          >
            <WithStyles(TableCell)>
              Romelu Lukaku
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Centre-Forward
            </WithStyles(TableCell)>
            <WithStyles(TableCell) />
          </WithStyles(TableRow)>
          <WithStyles(TableRow)
            className="PlayersTable--row"
          >
            <WithStyles(TableCell)>
              David de Gea
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Keeper
            </WithStyles(TableCell)>
            <WithStyles(TableCell) />
          </WithStyles(TableRow)>
        </WithStyles(TableBody)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(TableRow)>
            <WithStyles(TableCell)>
              Player
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Position
            </WithStyles(TableCell)>
            <WithStyles(TableCell)>
              Age
            </WithStyles(TableCell)>
          </WithStyles(TableRow)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(TableCell)>
                Player
              </WithStyles(TableCell)>,
              <WithStyles(TableCell)>
                Position
              </WithStyles(TableCell)>,
              <WithStyles(TableCell)>
                Age
              </WithStyles(TableCell)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Player",
              },
              "ref": null,
              "rendered": "Player",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Position",
              },
              "ref": null,
              "rendered": "Position",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Age",
              },
              "ref": null,
              "rendered": "Age",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(TableRow)
              className="PlayersTable--row"
            >
              <WithStyles(TableCell)>
                Romelu Lukaku
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Centre-Forward
              </WithStyles(TableCell)>
              <WithStyles(TableCell) />
            </WithStyles(TableRow)>,
            <WithStyles(TableRow)
              className="PlayersTable--row"
            >
              <WithStyles(TableCell)>
                David de Gea
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Keeper
              </WithStyles(TableCell)>
              <WithStyles(TableCell) />
            </WithStyles(TableRow)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Romelu LukakuCentre-Forward",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableCell)>
                  Romelu Lukaku
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)>
                  Centre-Forward
                </WithStyles(TableCell)>,
                <WithStyles(TableCell) />,
              ],
              "className": "PlayersTable--row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Romelu Lukaku",
                },
                "ref": null,
                "rendered": "Romelu Lukaku",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Centre-Forward",
                },
                "ref": null,
                "rendered": "Centre-Forward",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "David de GeaKeeper",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableCell)>
                  David de Gea
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)>
                  Keeper
                </WithStyles(TableCell)>,
                <WithStyles(TableCell) />,
              ],
              "className": "PlayersTable--row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "David de Gea",
                },
                "ref": null,
                "rendered": "David de Gea",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Keeper",
                },
                "ref": null,
                "rendered": "Keeper",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(TableHead)>
            <WithStyles(TableRow)>
              <WithStyles(TableCell)>
                Player
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Position
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Age
              </WithStyles(TableCell)>
            </WithStyles(TableRow)>
          </WithStyles(TableHead)>,
          <WithStyles(TableBody)>
            <WithStyles(TableRow)
              className="PlayersTable--row"
            >
              <WithStyles(TableCell)>
                Romelu Lukaku
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Centre-Forward
              </WithStyles(TableCell)>
              <WithStyles(TableCell) />
            </WithStyles(TableRow)>
            <WithStyles(TableRow)
              className="PlayersTable--row"
            >
              <WithStyles(TableCell)>
                David de Gea
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Keeper
              </WithStyles(TableCell)>
              <WithStyles(TableCell) />
            </WithStyles(TableRow)>
          </WithStyles(TableBody)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(TableRow)>
              <WithStyles(TableCell)>
                Player
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Position
              </WithStyles(TableCell)>
              <WithStyles(TableCell)>
                Age
              </WithStyles(TableCell)>
            </WithStyles(TableRow)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableCell)>
                  Player
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)>
                  Position
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)>
                  Age
                </WithStyles(TableCell)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Player",
                },
                "ref": null,
                "rendered": "Player",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Position",
                },
                "ref": null,
                "rendered": "Position",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Age",
                },
                "ref": null,
                "rendered": "Age",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(TableRow)
                className="PlayersTable--row"
              >
                <WithStyles(TableCell)>
                  Romelu Lukaku
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  Centre-Forward
                </WithStyles(TableCell)>
                <WithStyles(TableCell) />
              </WithStyles(TableRow)>,
              <WithStyles(TableRow)
                className="PlayersTable--row"
              >
                <WithStyles(TableCell)>
                  David de Gea
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  Keeper
                </WithStyles(TableCell)>
                <WithStyles(TableCell) />
              </WithStyles(TableRow)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Romelu LukakuCentre-Forward",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableCell)>
                    Romelu Lukaku
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell)>
                    Centre-Forward
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell) />,
                ],
                "className": "PlayersTable--row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Romelu Lukaku",
                  },
                  "ref": null,
                  "rendered": "Romelu Lukaku",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Centre-Forward",
                  },
                  "ref": null,
                  "rendered": "Centre-Forward",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "David de GeaKeeper",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableCell)>
                    David de Gea
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell)>
                    Keeper
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell) />,
                ],
                "className": "PlayersTable--row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "David de Gea",
                  },
                  "ref": null,
                  "rendered": "David de Gea",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Keeper",
                  },
                  "ref": null,
                  "rendered": "Keeper",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
